tests/dream/geant4_reduction_test.py

```diff
@@ -47,77 +48,76 @@ def params(request):
+    return wf
 
 
-def test_pipeline_can_compute_dspacing_result(providers, params):
-    pipeline = sciline.Pipeline(providers, params=params)
-    pipeline = powder.with_pixel_mask_filenames(pipeline, [])
-    result = pipeline.compute(IofDspacing)
+def test_pipeline_can_compute_dspacing_result(workflow):
+    workflow = powder.with_pixel_mask_filenames(workflow, [])
+    result = workflow.compute(IofDspacing)
     assert result.sizes == {
-        'dspacing': len(params[DspacingBins]) - 1,
+        'dspacing': len(workflow.compute(DspacingBins)) - 1,
     }
-    assert sc.identical(result.coords['dspacing'], params[DspacingBins])
+    assert sc.identical(result.coords['dspacing'], workflow.compute(DspacingBins))
 
 
-def test_workflow_is_deterministic(providers, params):
-    pipeline = sciline.Pipeline(providers, params=params)
-    pipeline = powder.with_pixel_mask_filenames(pipeline, [])
+def test_workflow_is_deterministic(workflow):
+    workflow = powder.with_pixel_mask_filenames(workflow, [])
     # This is Sciline's default scheduler, but we want to be explicit here
     scheduler = sciline.scheduler.DaskScheduler()
-    graph = pipeline.get(IofDspacing, scheduler=scheduler)
+    graph = workflow.get(IofDspacing, scheduler=scheduler)
     reference = graph.compute().data
     result = graph.compute().data
     assert sc.identical(sc.values(result), sc.values(reference))
 
 
-def test_pipeline_can_compute_intermediate_results(providers, params):
-    pipeline = sciline.Pipeline(providers, params=params)
-    pipeline = powder.with_pixel_mask_filenames(pipeline, [])
-    result = pipeline.compute(NormalizedByProtonCharge[SampleRun])
+def test_pipeline_can_compute_intermediate_results(workflow):
+    workflow = powder.with_pixel_mask_filenames(workflow, [])
+    result = workflow.compute(NormalizedByProtonCharge[SampleRun])
     assert set(result.dims) == {'segment', 'wire', 'counter', 'strip', 'module'}
 
 
-def test_pipeline_group_by_two_theta(providers, params):
-    params[TwoThetaBins] = sc.linspace(
+def test_pipeline_group_by_two_theta(workflow):
+    workflow[TwoThetaBins] = sc.linspace(
         dim='two_theta', unit='rad', start=0.8, stop=2.4, num=17
     )
-    pipeline = sciline.Pipeline(providers, params=params)
-    pipeline = powder.with_pixel_mask_filenames(pipeline, [])
-    result = pipeline.compute(IofDspacingTwoTheta)
+    workflow = powder.with_pixel_mask_filenames(workflow, [])
+    result = workflow.compute(IofDspacingTwoTheta)
     assert result.sizes == {
         'two_theta': 16,
-        'dspacing': len(params[DspacingBins]) - 1,
+        'dspacing': len(workflow.compute(DspacingBins)) - 1,
     }
-    assert sc.identical(result.coords['dspacing'], params[DspacingBins])
-    assert sc.allclose(result.coords['two_theta'], params[TwoThetaBins])
+    assert sc.identical(result.coords['dspacing'], workflow.compute(DspacingBins))
+    assert sc.allclose(result.coords['two_theta'], workflow.compute(TwoThetaBins))
```

Comment:
I prefer the old implementation because if checks that the workflow respects the parameters. The new tests only check that the workflow's outputs are consistent.

Reply:
Refactored, see update.